### PR TITLE
Update repository files endpoint for APIv4

### PR DESCRIPTION
**Pull request by mltsy**
New path for all endpoints is `repository/files/:file_path` (instead of passing file_path as a parameter)

### CODE REVIEW

File: lib/gitlab/client/repository_files.rb [hunk 1]
```diff
@@ -51,9 +51,8 @@ def get_file(project, file_path, ref)
     # @param  [String] commit message.
     # @return [Gitlab::ObjectifiedHash]
     def create_file(project, path, branch, content, commit_message)
```
Comment (asedge) [inline lib/gitlab/client/repository_files.rb]: Looking at the docs there seem to be a bunch of optional parameters we don't currently support.  Do you think we should add `options={}` and update the YARD to include the optional parameters defined in the Gitlab API doc for `Repository Files`?

File: lib/gitlab/client/repository_files.rb [hunk 2]
```diff
@@ -70,9 +69,8 @@ def create_file(project, path, branch, content, commit_message)
     # @param  [String] commit message.
     # @return [Gitlab::ObjectifiedHash]
     def edit_file(project, path, branch, content, commit_message)
```
Comment (asedge) [inline lib/gitlab/client/repository_files.rb]: Looking at the docs there seem to be a bunch of optional parameters we don't currently support. Do you think we should add options={} and update the YARD to include the optional parameters defined in the Gitlab API doc for Repository Files?

File: lib/gitlab/client/repository_files.rb [hunk 3]
```diff
@@ -88,9 +86,8 @@ def edit_file(project, path, branch, content, commit_message)
     # @param  [String] commit message.
     # @return [Gitlab::ObjectifiedHash]
     def remove_file(project, path, branch, commit_message)
```
Comment (asedge) [inline lib/gitlab/client/repository_files.rb]: Looking at the docs there seem to be a bunch of optional parameters we don't currently support. Do you think we should add options={} and update the YARD to include the optional parameters defined in the Gitlab API doc for Repository Files?

**Comment by mltsy** [comment]
That should do it! :)

**Comment by asedge** [comment]
Thanks! 👍

**Pull request merged**